### PR TITLE
added allow list functionality to analyzer engine

### PR DESCRIPTION
**Pull request by akshara-msft**
## Change Description

Added functionally to add an allow list of words that should be allowed in the text and are specified to not be PII. Includes unit tests and additional documentation using example. 

## Issue reference

This PR fixes issue: not specified on github

## Checklist

- [x] I have reviewed the [contribution guidelines](https://github.com/microsoft/presidio/blob/main/CONTRIBUTING.md)
- [x] I have signed the CLA
- [x] My code includes unit tests
- [x] All unit tests and lint checks pass locally
- [x] My PR contains documentation updates / additions if required

### CODE REVIEW

**Comment by omri374** [comment]
Thanks! Great work, some minor comments but overall great work.

File: presidio-analyzer/presidio_analyzer/analyzer_engine.py [hunk 1]
```diff
@@ -48,6 +48,7 @@ def __init__(
         default_score_threshold: float = 0,
         supported_languages: List[str] = None,
         context_aware_enhancer: Optional[ContextAwareEnhancer] = None,
+        allow_list=None,
```
[the remark this answers is not in the document]
Comment (akshara-msft) [reply presidio-analyzer/presidio_analyzer/analyzer_engine.py:51]: no, removing

File: SECURITY.md [hunk 2]
```diff
@@ -1,20 +1,14 @@
-<!-- BEGIN MICROSOFT SECURITY.MD V0.0.7 BLOCK -->
```
Comment (omri374) [inline SECURITY.md:1]: This file should not be changed, not sure why It's ilke that.

**Review by omri374 — DISMISSED** [review]
Looks great! Thanks @akshara-msft

**Review by omri374 — APPROVED** [review]

**Review by navalev — APPROVED** [review]

**Pull request merged**